src/agent/task_manager/include/itask_manager.hpp

```diff
@@ -24,9 +25,11 @@ class ITaskManager
 
     /// @brief Enqueues a task to be executed
     /// @param task The task to enqueue
-    virtual void EnqueueTask(std::function<void()> task) = 0;
+    /// @param taskID The ID of the task
+    virtual void EnqueueTask(std::function<void()> task, const std::string& taskID) = 0;
```

Comment:
Why did you choose a string for this?

Reply:
I thought it was the easiest to set up from a user point of view, but any suggestion is welcome. What I wanted it to be able to know which task threw an exception, which is not possible to do without an identifier.

Reply:
I agree, let's keep this.